beamer/check/commands.py

```diff
@@ -0,0 +1,190 @@
+import json
+import random
+import sys
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import apischema
+import click
+import structlog
+from apischema import schema
+from eth_utils import to_checksum_address
+from hexbytes import HexBytes
+from web3.constants import ADDRESS_ZERO
+from web3.contract import Contract, ContractConstructor
+from web3.contract.contract import ContractFunction
+from web3.types import TxReceipt
+
+import beamer.contracts
+import beamer.util
+from beamer.contracts import ABIManager, obtain_contract
+from beamer.typing import ChainId, FillId, RequestId, TokenAmount
+from beamer.util import ChainIdParam, create_request_id, make_web3
+
+log = structlog.get_logger(__name__)
+
+
+@dataclass
+class Invalidation:
+    proof_source: ChainId = field(metadata=schema(min=1))
+    proof_target: ChainId = field(metadata=schema(min=1))
+    request_id: RequestId
+    fill_id: FillId
+    txhash: HexBytes
+    finalization_timestamp: datetime
+
+
+def _transact(func: ContractConstructor | ContractFunction) -> TxReceipt:
+    try:
+        receipt = beamer.util.transact(func)
+    except beamer.util.TransactionFailed as exc:
+        log.error("Transaction failed", exc=exc)
+        sys.exit(1)
```

Comment:
What's the motivation behind the strict exit here?

Reply:
Two things:
- consistency with the other command(s)
- "fail early and complain loudly" so the user has a much higher chance of noticing that something went wrong